### DISCUSSION TITLE
Using a SolutionUserObject with the Thermal Hydraulics Module

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#25713] [Q&A Modules: General] [answered]
Hi MOOSE team,
I have encountered a problem that occurs when using a `SolutionUserObject` in a thermal hydraulics problem.  I am using a `SolutionUserObject` to read in a volumetric heating variable, aptly named `vol_heating`, that is calculated in another MOOSE app. I then use this term with a `HeatSourceFromPowerDensity` object in order to apply the heating term on my heat structure. This works perfectly fine when I do not have any flow channels defined, the heat structure heats up as expected. However, when channels are defined, I get an error of the following sort.

```
The following error occurred in the object "solution_in", of type "SolutionUserObject".

Failed to access the data for variable 'vol_heating' at point (x,y,z)=(       0,        0,       -5) in the 'solution_in' SolutionUserObject
```

The coordinate 0, 0, -5 are the starting coordinates of one of my flow channels. The `SolutionUserObject` is looking for the variable value on my flow channels. I understand why, but is there any way to prevent this from happening? I have tried using 'execute_on = INITIAL', but to no success. I have included some input files and a mesh which together constitute a minimum working example. The volumetric_heating input file needs to be run first so that the results can be imported in the thermal hydraulics input files with the SolutionUserObject. Running the `th` input file will result in the same error I mentioned above. The `no_channels_th` input file is almost identical, but has the flow channel removed, and runs just fine. 

 To summarise, I want to read in a variable that only exists on the heat structure, not the flow channels. 

Any help is greatly appreciated:) 

[volumetric_heating.txt](https://github.com/idaholab/moose/files/12870176/volumetric_heating.txt)
[no_th_channels.txt](https://github.com/idaholab/moose/files/12870182/no_th_channels.txt)
[th.txt](https://github.com/idaholab/moose/files/12870184/th.txt)
[Mesh.zip](https://github.com/idaholab/moose/files/12870217/Mesh.zip)

**Comment by GiudGiud** [comment, marked as answer]
I ll take a look

EDIT Ignore my previous reply, wrong post

Guillaume

**Reply by GiudGiud** [reply]
actually it wasnt that bad

you just need to add a block restriction 

```
[AuxKernels]
  [aux-vol_heating]
    type = SolutionAux
    solution = solution_in
    variable = vol_heating
    from_variable = vol_heating
    scale_factor = 1e-2

    block = 'hs_geom:theblock'
  []
[]
```

**Reply by TheBEllis** [reply]
Thanks Guillaume:)
